In [2]:
# First, install the required libraries using pip
!pip install sqlalchemy mysql-connector-python

In [3]:
# Import necessary libraries
import pandas as pd
from sqlalchemy import create_engine, text, select

In [4]:
# Database connection details
db_username = 'root'
db_password = '11111111'
db_host = 'localhost' 
db_port = '3306'       
db_database = 'LibraryDB'

In [5]:
# Create a SQLAlchemy engine to connect to the MySQL database
engine = create_engine(f'mysql+mysqlconnector://{db_username}:{db_password}@{db_host}:{db_port}/{db_database}')

# Books 

In [6]:
# Create a SQLAlchemy select statement for the table
books = text("SELECT * FROM Books")

In [7]:
# Execute the select statement and fetch the data into a DataFrame
with engine.connect() as connection:
    result = connection.execute(books)
    books_df = pd.DataFrame(result.fetchall(), columns=result.keys())

In [8]:
# Print the DataFrame containing the fetched data
print(books_df)

   BookID                   Title ISBN  PublicationYear  AuthorID
0     101  The Art of Programming  977             2022         1
1     102     Data Science Basics  978             2021         2
2     103     Web Development 101  979             2023         1


In [9]:
# Display the DataFrame in a Jupyter Notebook or compatible environment
display(books_df)

BookID                   Title ISBN  PublicationYear  AuthorID
0     101  The Art of Programming  977             2022         1
1     102     Data Science Basics  978             2021         2
2     103     Web Development 101  979             2023         1

# Authors

In [10]:
# Create a SQLAlchemy select statement for the table
authors = text("SELECT * FROM Authors ")

In [11]:
# Execute the select statement and fetch the data into a DataFrame
with engine.connect() as connection:
    result = connection.execute(authors)
    authors_df = pd.DataFrame(result.fetchall(), columns=result.keys())

In [12]:
# Print the DataFrame containing the fetched data
print(authors_df)

   AuthorID FirstName LastName   BirthDate
0         1      John      Doe  1970-05-15
1         2      Jane    Smith  1985-03-22


In [13]:
# Display the DataFrame in a Jupyter Notebook or compatible environment
display(authors_df)

AuthorID FirstName LastName   BirthDate
0         1      John      Doe  1970-05-15
1         2      Jane    Smith  1985-03-22

MySQL Script to Writge Before Running the Code :-

-- Create the database
CREATE DATABASE IF NOT EXISTS LibraryDB;

-- Use the Library database
USE LibraryDB;

-- Create the Authors table
CREATE TABLE IF NOT EXISTS Authors (
    AuthorID INT PRIMARY KEY,
    FirstName VARCHAR(255),
    LastName VARCHAR(255),
    BirthDate DATE
);

-- Create the Books table with a foreign key constraint referencing Authors table
CREATE TABLE IF NOT EXISTS Books (
    BookID INT PRIMARY KEY,
    Title VARCHAR(255),
    ISBN VARCHAR(255),
    PublicationYear INT,
    AuthorID INT,
    FOREIGN KEY (AuthorID) REFERENCES Authors(AuthorID)
);

-- Insert data into Authors table
INSERT INTO Authors (AuthorID, FirstName, LastName, BirthDate)
VALUES
    (1, 'John', 'Doe', '1970-05-15'),
    (2, 'Jane', 'Smith', '1985-03-22');

-- Insert data into Books table
INSERT INTO Books (BookID, Title, ISBN, PublicationYear, AuthorID)
VALUES
    (101, 'The Art of Programming', '977', 2022, 1),
    (102, 'Data Science Basics', '978', 2021, 2),
    (103, 'Web Development 101', '979', 2023, 1);